### DISCUSSION TITLE
Local-Only Instance w/ Caddy and HTTPS on Synology

### DISCUSSION THREAD

**Discussion by colin19889** [dani-garcia/vaultwarden#2113] [Q&A]
I'm attempting to run a private instance (not accessible outside of the LAN) of VW on my Synology. I can get it up and running but have issues doing anything with the web interface because it's not accessible via https. I've tried following [this guide](https://github.com/dani-garcia/vaultwarden/wiki/Running-a-private-vaultwarden-instance-with-Let%27s-Encrypt-certs ) and using Caddy but being that it's not written for Synology devices, and that I'm rather new to Docker Compose, there's probably something I'm missing.

My docker-compose.yml and Caddyfile consist of the following. The output says that both Dockers were created successfully and I can see both of them in the Docker GUI. I can access VM via port 7777 but don't seem to have any access to Caddy. I downloaded a custom build for Duck DNS for which I already have an account and manually changed the IP for that domain to my local Synology IP.

`version: '3'

services:
  vaultwarden:
    image: vaultwarden/server:latest
    container_name: VaultWarden
    restart: always
    environment:
      - WEBSOCKET_ENABLED=true
    ports:
      - 7777:80
      - 7776:3012
    volumes:
      - /volume2/docker/VaultWarden/vw-data:/data

  caddy:
    image: caddy:2
    container_name: Caddy
    restart: always
    ports:
      - 7779:80
      - 7778:443
    volumes:
      - /volume2/docker/VaultWarden/caddy:/usr/bin/caddy  # Your custom build of Caddy.
      - /volume2/docker/VaultWarden/Caddyfile:/etc/caddy/Caddyfile:ro
      - /volume2/docker/VaultWarden/caddy-config:/config
      - /volume2/docker/VaultWarden/caddy-data:/data
    environment:
      - DOMAIN=https://myvw.duckdns.org
      - EMAIL=myemail@gmail.com
      - DUCKDNS_TOKEN=12345678-1234-1234-1234-123456789123
      - LOG_FILE=/data/access.log`

My Caddyfile consists of the following,

`{$DOMAIN}:443 {
  log {
    level INFO
    output file {$LOG_FILE} {
      roll_size 10MB
      roll_keep 10
    }
  }

  tls {
    dns duckdns {$DUCKDNS_TOKEN}
  }

  encode gzip

  reverse_proxy /notifications/hub vaultwarden:3012

  reverse_proxy vaultwarden:80`

**Comment by Banshee350** [comment]
You may have to set an exception for DNS rebind protection in your router (e.g. FritzBox), otherwise domains with local ip will not be resolved and get blocked.
